python/doc/user_manual/base_objects.rst

```diff
@@ -94,10 +101,21 @@ Vectors
     Description
     Indices
     IndicesCollection
+    UnsignedIntegerCollection
     Point
     PointWithDescription
     ComplexCollection
     ScalarCollection
+    FunctionCollection
+    PolynomialCollection
+    OptimizationResultCollection
```

Comment:
About all these collections: maybe we should revise the doc of the "Collection" page to remove all the examples and simply say these are underlying C++ objects which should not be directly used from the Python API.

Reply:
the problem is that if we want to actually document the actual type returned we need these objects to be referenced somewhere, but else we could also say "sequence of XXX" or "list of XXX" for these